### PR TITLE
Update PR template to catch common mistakes

### LINKED ISSUE #2929: PR template is short on guidance, can trip up CLI users
#### Problem
CHIP's PR template is often ignored or when used from the CLI tools, loses markdown.  The "fixes" section seems a source or particular confusion, as most misinterpret what little guidance is offered.

 #### Proposed Solution
add moar guidance

### PR DESCRIPTION
**Pull request by rwalker-apple**
#### Problem
 Our PR template gives little guidance to first-time contributors and can confuse CLI users.

 #### Summary of Changes
 Add more guidance for Problem, Summary, and Fixes sections

 Fixes #2929

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,7 +1,60 @@
+<!-- ----------------------------------------------------------------
+  If you're editing this as a result of an invocation of a GitHub CLI
+   tool, note that lines that begin with '#' are stripped. To preserve the
+   markdown that begins with '#' below, be sure to preserve the leading
+   whitespace on those lines.
```
Comment (gerickson) [inline .github/PULL_REQUEST_TEMPLATE.md:5]: Interesting, I always end up deleting that leading white space since it always struck me as a typo.

**Review by gerickson — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (rwalker-apple) [reply .github/PULL_REQUEST_TEMPLATE.md:5]: yeap, lots of folks did.  you're in good company

**Review by andy31415 — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

**Review by mspang — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]
This seems good in general.  One issue I have with the whole setup is that it seems like a lot of the filled-out PR template should really go in the commit messages; in the case of single-commit PRs it basically duplicates the commit message.  I'm not sure what to do about that, though.

**Comment by bzbarsky-apple** [comment]
Maybe we should explicitly point out in the PR template that filling it out does not obviate the need for good commit messages?

**Comment by rwalker-apple** [comment]
> Maybe we should explicitly point out in the PR template that filling it out does not obviate the need for good commit messages?

I'd love to get a @bzbarsky-apple rant on commit messages in here somewhere.  Too bad commit message templates have to be added by the dev in their own tree..

**Pull request merged**